From: jbeulich@suse.com
Subject: fix pCPU handling
Patch-mainline: Never, SUSE-Xen specific
References: luckily none so far

Introduction of
patches.fixes/0017-acpi-processor-introduce-apic_id-in-struct-processor-to-save-parsed-apic-id
required dropping some changes done in xen3-patch-2.6.28 and
xen3-patch-3.3. While doing so I overlooked the need to also adjust
callers of acpi_get_cpuid() accordingly.

--- a/arch/x86/kernel/acpi/processor_extcntl_xen.c
+++ b/arch/x86/kernel/acpi/processor_extcntl_xen.c
@@ -212,7 +212,7 @@ static int xen_hotplug_notifier(struct a
 		return -EOPNOTSUPP;
 	}
 
-	apic_id = acpi_get_cpuid(pr->handle, ~device_decl, pr->acpi_id);
+	apic_id = acpi_get_apicid(pr->handle, device_decl, pr->acpi_id);
 	if (apic_id == -1) {
 		pr_warn("can't get apic_id for acpi_id %#x\n", pr->acpi_id);
 		return -ENODATA;
